tidy3d/plugins/design/method.py

```diff
@@ -175,28 +110,466 @@ class MethodGrid(MethodIndependent):
+            self._handle_param_convert(invert_param_converter, [next_point])
+            opt.register(params=next_point, target=next_out)
+
+        # Handle subsequent iterations sequentially as BayOpt package does not allow for batched non-random predictions
+        for _ in range(self.n_iter):
+            next_point = opt.suggest(utility)
+            self._force_int(next_point, parameters)
+            self._handle_param_convert(param_converter, [next_point])
+            next_out, aux_out = self._extract_output(run_fn([next_point]))
+            result.append(next_out[0])
+            self._flatten_and_append(aux_out, total_aux_out)
+            self._handle_param_convert(invert_param_converter, [next_point])
+            opt.register(params=next_point, target=next_out[0])
+
+        if console is not None:
+            console.log(
+                f"Best Result: {opt.max['target']}\n"
+                f"Best Parameters: {' '.join([f'{param}: {val}' for param, val in opt.max['params'].items()])}\n"
+            )
+
+        # Output fn_args from the BO.opt object - getting results in situ as opt changes type to float
+        fn_args = []
+        for output in opt.res:
+            fn_args.append(output["params"])
+
+        return fn_args, result, total_aux_out, opt
+
+
+class MethodGenAlg(MethodOptimize, ABC):
+    """A standard method for performing genetic algorithm search"""
+
+    # Args for the user
+    solutions_per_pop: pd.PositiveInt = pd.Field(
+        ...,
+        title="Number of solutions per population.",
+        description="The number of solutions to be generated for each population.",
+    )
+
+    n_generations: pd.PositiveInt = pd.Field(
+        ...,
+        title="Number of generations.",
+        description="The maximum number of generations to run the genetic algorithm.",
+    )
+
+    n_parents_mating: pd.PositiveInt = pd.Field(
+        ...,
+        title="Number of parents mating.",
+        description="The number of solutions to be selected as parents for the next generation.",
+    )
+
+    stop_criteria: Optional[pd.constr(regex=r"\b(?:reach|saturate)_\d+\b")] = pd.Field(
+        default=None,
+        title="Early stopping criteria.",
+        description="Define the early stopping criteria. Supported words are 'reach_X' or 'saturate_X' where X is a number. See PyGAD docs for more details.",
+    )
+
+    parent_selection_type: Optional[
+        Literal["sss", "rws", "sus", "rank", "random", "tournament"]
+    ] = pd.Field(
+        default="sss",
+        title="Parent selection type.",
+        description="The style of parent selector. See the PyGAD docs for more details.",
+    )
+
+    crossover_type: Optional[Literal["single_point", "two_points", "uniform", "scattered"]] = (
+        pd.Field(
+            default="single_point",
+            title="Crossover type.",
+            description="The style of crossover operation. See the PyGAD docs for more details.",
```

Comment:
consider linking to pyGAD docs here and in other instances.

Reply:
Yeah I'd left it out here because the pygad docs are a total mess, everything important is essentially on one giant landing page **https://pygad.readthedocs.io/en/latest/pygad.html**. I guess it's more convenient than having the user Google it though.